SIMPLE/src/edu/manipal/icas/simple/impl/views/PassportGrantingOfficerDashboardViewImpl.java

```diff
@@ -0,0 +1,560 @@
+	private JComboBox<String> applicationIdComboBox;
+	private JLabel officerIdLabel;
+	private JButton logoutButton;
+	private JButton approveButton;
+	private JPanel navigationBarPanel;
+	private JButton rejectButton;
+	private JPanel containerPanel;
+	private JButton requestPoliceVerificationButton;
+	private JLabel nameLabel;
+	private JLabel genderLabel;
+	private JLabel emailAddressLabel;
+	private JLabel dateOfBirthLabel;
+	private JLabel nameOfMotherLabel;
+	private JLabel nameOfFatherLabel;
+	private JLabel birthAddressLabel;
+	private JLabel permanentAddressLabel;
+	private JLabel presentAddressLabel;
+	private JLabel contactNumberLabel;
+	public SwingController pdfBiometricsController;
+	public SwingController pdfDocumentsController;
+	/**
+	 * Initialises all the required components that are to be displayed in the PGO
+	 * dashboard view.
+	 */
+	public PassportGrantingOfficerDashboardViewImpl() {
+		super("Passport Granting Officer");
+		previousApplicationIdButton = new JButton(getScaledImage(ResourceConstants.IMAGE_PREVIOUS_BUTTON_ICON, 15, 15));
+		String testData[] = { "000000", "000001" };
+		applicationIdComboBox = new JComboBox<>(testData);
+		nextApplicationIdButton = new JButton(getScaledImage(ResourceConstants.IMAGE_NEXT_BUTTON_ICON, 15, 15));
+		officerIdLabel = new JLabel();
+		logoutButton = new JButton("Logout");
+		approveButton = new JButton("Approve");
+		rejectButton = new JButton("Reject");
+		tabbedPane = new JTabbedPane();
+		navigationBarPanel = new JPanel();
+		containerPanel = new JPanel();
+		requestPoliceVerificationButton = new JButton("Request police verification");
+		nameLabel = new JLabel();
+		genderLabel = new JLabel();
+		emailAddressLabel = new JLabel();
+		dateOfBirthLabel = new JLabel();
+		nameOfMotherLabel = new JLabel();
+		nameOfFatherLabel = new JLabel();
+		contactNumberLabel = new JLabel();
+		birthAddressLabel = new JLabel();
+		permanentAddressLabel = new JLabel();
+		presentAddressLabel = new JLabel();
+		pdfDocumentsController = new SwingController();
+		pdfBiometricsController = new SwingController();
+
+		initialiseUi();
+	}
+
+	/**
+	 * Gets the scaled version of the image icon passed to it.
+	 * 
+	 * @param icon   the image icon that is to be scaled to the desired size
+	 * @param width  the width that the image should be scaled to
+	 * @param height the height that the image should be scaled to
+	 * @return the scaled image
+	 */
+	private ImageIcon getScaledImage(ImageIcon icon, int width, int height) {
+		int newWidth = icon.getIconWidth();
+		int newHeight = icon.getIconHeight();
+
+		if (newWidth > width) {
+			newWidth = width;
+			newHeight = (newWidth * icon.getIconHeight()) / icon.getIconWidth();
+		}
+
+		if (newHeight > height) {
+			newHeight = height;
+			newWidth = (icon.getIconWidth() * newHeight) / icon.getIconHeight();
+		}
+
+		return new ImageIcon(icon.getImage().getScaledInstance(newWidth, newHeight, Image.SCALE_DEFAULT));
+	}
```

Comment:
Put this into a util class as it's being used in multiple views

Reply:
sure